File: Models/CatalogoModels.cs
namespace Softawer.Models;

public class Maquina
{
    public int IdMaquina { get; set; }
    public int IdTipoMaquina { get; set; }
    public string TipoMaquina { get; set; } = string.Empty;
    public string Nombre { get; set; } = string.Empty;
    public string Descripcion { get; set; } = string.Empty;
    public string MusculosObjetivo { get; set; } = string.Empty;
    public string ImagenUrl { get; set; } = string.Empty;
    public string Ubicacion { get; set; } = string.Empty;
    public string Estado { get; set; } = string.Empty;
    public int Cantidad { get; set; }
}

public class Ejercicio
{
    public int IdEjercicio { get; set; }
    public string Nombre { get; set; } = string.Empty;
    public string Descripcion { get; set; } = string.Empty;
    public string GrupoMuscular { get; set; } = string.Empty;
    public string Dificultad { get; set; } = string.Empty;
    public int? IdMaquina { get; set; }
    public string? NombreMaquina { get; set; }
}

File: Frontend_Software/src/app/core/models/catalogo.models.ts
export type Maquina = {
  idMaquina: number; idTipoMaquina: number; tipoMaquina: string; nombre: string; descripcion: string;
  musculosObjetivo: string; imagenUrl: string; ubicacion: string; estado: string; cantidad: number;
};
export type MaquinaFavoritaPayload = { idUsuario: number };
export type Ejercicio = {
  idEjercicio: number; nombre: string; descripcion: string; grupoMuscular: string; dificultad: string;
  idMaquina?: number | null; nombreMaquina?: string | null;
};
